packages/core/src/routes/swagger.ts

```diff
@@ -124,21 +130,31 @@ export default function swaggerRoutes<T extends AnonymousRouter, R extends Route
       )
     );
 
-    const pathMap = new Map<string, MethodMap>();
+    const paths = new Map<string, MethodMap>();
 
     // Group routes by path
     for (const { path, method, operation } of routes) {
-      pathMap.set(path, { ...pathMap.get(path), [method]: operation });
+      paths.set(path, { ...paths.get(path), [method]: operation });
     }
 
-    const document: OpenAPIV3.Document = {
-      openapi: '3.0.1',
-      info: {
-        title: 'Logto Core',
-        version: '0.1.0',
-      },
-      paths: Object.fromEntries(pathMap),
-    };
+    const additionalSwagger = load(
+      readFileSync('static/yaml/additional-swagger.yaml', 'utf-8')
+    ) as OpenAPIV3.Document;
+
+    const document: OpenAPIV3.Document = deepmerge(
+      { paths: Object.fromEntries(paths) },
+      additionalSwagger,
+      {
+        customMerge: (key) => {
+          // Overwrite OpenAPIV3.OperationObject.responses
+          // with the custom content from the additional swagger.
+          if (key === 'responses') {
```

Comment:
will it be better to use `for` instead of `deepmerge`? i don;t see the benefit here

Reply:
refactored.

append the additional `responses` to `OperationObject` when building `OperationObject`.